src/core/load-fragment.c

```diff
@@ -1102,12 +1104,18 @@ int config_parse_socket_bindtodevice(
                 return 0;
         }
 
-        if (!ifname_valid(rvalue)) {
-                log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid interface name, ignoring: %s", rvalue);
+        r = unit_name_printf(UNIT(s), rvalue, &p);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r, "Failed to resolve unit specifiers in %s, ignoring: %m", rvalue);
+                return 0;
+        }
+
+        if (!ifname_valid(p)) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid interface name, ignoring: %s", p);
                 return 0;
         }
 
-        return free_and_strdup_warn(&s->bind_to_device, rvalue);
+        return free_and_strdup_warn(&s->bind_to_device, p);
```

Comment:
free_and_replace

Reply:
Hm, looks like free_and_replace doesn't exactly fit to work with this first argument.

Reply:
Yeah, free_and_replace internally takes pointer of first arg, so the & here needs to be dropped